bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gnirs/InstGNIRS.java

```diff
@@ -222,10 +242,27 @@ public double getSetupTime(ISPObservation obs) {
         // REL-437: When GNIRS is used with Altair in LGS mode then the acq overhead must be 25 minutes.
         // The only change is to GNIRS with Altair LGS, for other modes the overhead should remain 15 minutes.
         if (altair != null && altair.getMode().guideStarType() == AltairParams.GuideStarType.LGS) {
-            return 25 * 60;
+            return getSetupTimeLgs();
         }
 
-        return (_slitWidth == SlitWidth.IFU) ? (20 * 60) : (15 * 60);
+        return (_slitWidth == SlitWidth.IFU) ? (getSetupTimeIfu()) : (getSetupTime());
+    }
+
+    public double getSetupTime(Config conf) {
+        String aoSystem = (String) conf.getItemValue(AOConstants.AO_SYSTEM_KEY);
+        String guideStarType = (String) conf.getItemValue(AOConstants.AO_GUIDE_STAR_TYPE_KEY);
+        String slitWidth = (String) conf.getItemValue(GNIRSConstants.SLIT_WIDTH_KEY);
```

Comment:
This should be a `GNIRSParams.SlitWidth` I believe, which can be checked to see whether it equals `GnirsParams.SlitWidth.IFU`.

Reply:
Replaced all Strings with the typed values from the model.